addons/im_livechat/static/src/models/discuss/discuss.js

```diff
@@ -0,0 +1,13 @@
+/** @odoo-module **/
+
+import { registerFieldPatchModel } from '@mail/model/model_core';
+import { one2one } from '@mail/model/model_field';
+
+registerFieldPatchModel('mail.discuss', 'im_livechat/static/src/models/discuss/discuss.js', {
+    /**
+     * Discuss sidebar category for `livechat` channel threads.
+     */
+    categoryLivechat: one2one('mail.discuss_sidebar_category', {
+        isCausal: true,
+    }),
```

Comment:
As we have discussed earlier before, I think it may tricky to make it `required` and `readonly`
- for `readonly`: the category needs to be updated with server dynamically (e.g. open/close category)
- for `required`: most of methods and fields depend on `env.messaging` and `env.messaging.discuss`. It has to be initialized after `messaging`